File: Backend/StudyPlatform/Models/DTOs/GeneratedMindmapDTO.cs
using StudyPlatform.Data.Models;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace StudyPlatform.Models.DTOs
{
    public class GeneratedMindmapDTO
    {
        [JsonPropertyName("nodes")]
        [Required]
        public List<MindmapNodeDTO> Nodes { get; set; }

        [JsonPropertyName("edges")]
        [Required]
        public List<MindmapEdgeDTO> Edges { get; set; }

        [JsonPropertyName("title")]
        [Required]
        public string Title { get; set; }

        [JsonPropertyName("description")]
        [Required]
        public string Description { get; set; }

        [JsonPropertyName("difficulty")]
        [Required]
        public Difficulty Difficulty { get; set; }
    }
}

File: Frontend/src/data/DTOs/GeneratedMindmapDTO.ts
import { Difficulty } from "../Difficulty";

export interface GeneratedMindmapDTO {
  nodes: MindmapNodeDTO[];
  edges: MindmapEdgeDTO[];
  materialSubGroupId?: string | null;
  title: string;
  description: string;
  difficulty: Difficulty;
}

export interface MindmapNodeDTO {
  id: string;
  data: NodeDataDTO;
  position: NodePositionDTO;
}

export interface NodeDataDTO {
  label: string;
}

export interface NodePositionDTO {
  x: number;
  y: number;
}

export interface MindmapEdgeDTO {
  id: string;
  source: string;
  target: string;
  label: string;
}
